plugins/modules/proxmox_cluster.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2025, Florian Paul Azim Hoberg (@gyptazy) <florian.hoberg@credativ.de>
+#
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: proxmox_cluster
+version_added: 0.1.0
+short_description: Management of Proxmox Nodes for creating a Proxmox VE cluster
+description:
+  - Allows you to create/join a Proxmox VE cluster from an empty Proxmox VE node.
+author: Florian Paul Azim Hoberg (@gyptazy) <florian.hoberg@credativ.de>
+options:
+  api_host:
+    description:
+      - The hostname of the Proxmox VE node to perform actions.
+    type: str
+    required: true
+  api_user:
+    description:
+      - The username to use to login to the Proxmox VE API.
+    type: str
+    required: true
+  api_password:
+    description:
+      - The password to use to login to the Proxmox VE API.
+    type: str
+    required: true
+  api_ssl_verify:
+    description:
+      - Whether to verify the SSL certificate of the Proxmox VE API.
+    type: bool
+    required: false
+    default: true
+  link0:
+    description:
+      - The first IP address to use for cluster communication.
+    type: str
+    required: false
+  link1:
+    description:
+      - The second IP address to use for cluster communication.
+    type: str
+    required: false
+  master_ip:
+    description:
+      - The IP address of the cluster master when joining the cluster.
+    type: str
+    required: false
+  fingerprint:
+    description:
+      - The fingerprint of the cluster master when joining the cluster.
+    type: str
+    required: false
+  cluster_name:
+    description:
+      - The cluster name to use for cluster creation.
+    type: str
+    required: false
+  action:
+    description:
+      - Possible module actions to perform by the Proxmox VE API.
+    choices: ['create_cluster', 'join_cluster', "get_cluster_info"]
```

Comment:
Sorry I didn't notice this earlier, but: modules should not have **actions**, but **states**. Also, as @Thulium-Drake pointed out, `get_cluster_info` should be moved to an `_info` module. See https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html#designing-module-interfaces and https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html#scoping-your-module-s.